play.pokemonshowdown.com/js/client-teambuilder.js

```diff
@@ -1297,7 +1297,7 @@
 			buf += '<div class="setchart-nickname">';
 			buf += '<label>Nickname</label><input type="text" name="nickname" class="textbox" value="' + BattleLog.escapeHTML(set.name || '') + '" placeholder="' + BattleLog.escapeHTML(species.baseSpecies) + '" />';
 			buf += '</div>';
-			buf += '<div class="setchart" style="' + Dex.getTeambuilderSprite(set, this.curTeam.gen) + ';">';
+			buf += '<div class="setchart" style="' + Dex.getTeambuilderSprite(set, this.curTeam.dex.modid) + ';">';
```

Comment:
At that point you could just send the whole dex in, so you don't need to extract the gen from the modid.

Reply:
I'll do this myself.